application-analytics-ui/src/main/resources/Analytics/Code/Macros/MostViewedPages.xml

```diff
@@ -0,0 +1,812 @@
+        <size>10</size>
+        <unmodifiable>0</unmodifiable>
+        <classType>com.xpn.xwiki.objects.classes.NumberClass</classType>
+      </priority>
+      <supportsInlineMode>
+        <disabled>0</disabled>
+        <displayFormType>select</displayFormType>
+        <displayType>yesno</displayType>
+        <name>supportsInlineMode</name>
+        <number>5</number>
+        <prettyName>Supports inline mode</prettyName>
+        <unmodifiable>0</unmodifiable>
+        <classType>com.xpn.xwiki.objects.classes.BooleanClass</classType>
+      </supportsInlineMode>
+      <visibility>
+        <cache>0</cache>
+        <disabled>0</disabled>
+        <displayType>select</displayType>
+        <freeText>forbidden</freeText>
+        <largeStorage>0</largeStorage>
+        <multiSelect>0</multiSelect>
+        <name>visibility</name>
+        <number>6</number>
+        <prettyName>Macro visibility</prettyName>
+        <relationalStorage>0</relationalStorage>
+        <separator>|</separator>
+        <separators>|</separators>
+        <size>1</size>
+        <unmodifiable>0</unmodifiable>
+        <values>Current User|Current Wiki|Global</values>
+        <classType>com.xpn.xwiki.objects.classes.StaticListClass</classType>
+      </visibility>
+    </class>
+    <property>
+      <async_cached>0</async_cached>
+    </property>
+    <property>
+      <async_context/>
+    </property>
+    <property>
+      <async_enabled>0</async_enabled>
+    </property>
+    <property>
+      <code>{{velocity}}
+#set ($discard = $xwiki.jsx.use('Analytics.Code.Macros.MostViewedPages'))
+#set ($object =  $xwiki.getDocument("Analytics.Code.Configuration").getObject('Analytics.Code.ConfigurationClass'))
+#set ($ip = $object.getProperty("requestAddress").value)
+#set ($idSite = $object.getProperty("idSite").value)
+#set ($authToken = $object.getProperty("authToken").value)
+#set ($dashbord =  $doc.getObject('Analytics.Code.AnalyticsDashboardClass'))
+#set ($period = 'range')
+## A dashboard will have a date inside and that date will take precedence over the date set by the macro.
+#if ($dashbord)
+  #set ($startDate =  $dashbord.getProperty("startDate"))
+  #set ($endTimeDate =  $dashbord.getProperty("endDate"))
+#else
+  #set ($startDate = $xcontext.macro.params.startDate.toString())
+  #set ($endTimeDate = $xcontext.macro.params.endDate.toString())
+#end
+#set ($rows = $xcontext.macro.params.numberOfRows)
+#set ($startDate = $datetool.toDate('yyyy/MM/dd HH:MM', $startDate))
+#set ($endTimeDate = $datetool.toDate('yyyy/MM/dd', $endTimeDate))
+#set ($startDate = $datetool.format('yyyy-MM-dd HH:MM', $startDate))
+#set ($endTimeDate = $datetool.format('yyyy-MM-dd', $endTimeDate))
+#set ($date = $startDate + ','+$endTimeDate)
+## This will check if any either the startDate or the endDate is missing and replace the values to retrive the data for
+## the last year
+# if(!$startDate || !$endTimeDate)
+  #set($period = 'year')
+  #set($date = 'last1')
+#end
```

Comment:
This is not actually very accurate, since in case one of these values is defined, you are overriding them (removing any inferior or superior limit the user has set)
I think you should remove for this year filter. Use today as `$endDate` and a another date for `$startDate` (you had a suggestion the other day)